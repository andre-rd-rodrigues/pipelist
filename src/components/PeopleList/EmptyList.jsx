import React from "react";
import { motion } from "framer-motion";
import {
  containerVariant,
  horizontalEntranceLeft
} from "styles/motion/motionVariants";
import styles from "./list.module.scss";

function EmptyList() {
  return (
    <motion.div
      variants={containerVariant}
      initial="hidden"
      animate="visible"
      className={styles.emptyList}
    >
      <motion.img
        variants={horizontalEntranceLeft}
        src={`${process.env.PUBLIC_URL}/images/start.png`}
        alt="Start your pipelist!"
      />
      <motion.p variants={horizontalEntranceLeft}>
        Currently your list is empty. <b>Add a person</b> to start your very
        first pipelist!
      </motion.p>
    </motion.div>
  );
}

export default EmptyList;
