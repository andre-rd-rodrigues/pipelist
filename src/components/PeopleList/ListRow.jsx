import React from "react";
import OfficeIcon from "assets/icons/office_icon";
import Avatar from "components/Avatar/Avatar";
import { empty_field } from "utils/configs";
import styles from "./list.module.scss";

const ListRow = ({ onPersonSelect, person, provided }) => {
  const { innerRef, draggableProps, dragHandleProps } = provided;

  const onRowClicked = () => {
    onPersonSelect(person);
  };
  return (
    <div
      className={styles.listRow}
      role="button"
      onClick={onRowClicked}
      onKeyPress={onRowClicked}
      tabIndex={0}
      ref={innerRef}
      {...draggableProps}
      {...dragHandleProps}
      data-testid="list_row_div"
    >
      <div>
        <h4>{`${person?.first_name} ${person?.last_name}`}</h4>
        <div className="list_row_person_address_div">
          <OfficeIcon />
          <p>{person?.org_name || empty_field}</p>
        </div>
      </div>
      <Avatar src={person?.picture_id?.pictures[128]} alt="Pipelist avatar" />
    </div>
  );
};

export default ListRow;
