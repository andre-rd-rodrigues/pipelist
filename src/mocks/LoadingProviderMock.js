import { LoadingProvider } from "context/loading-context";

const LoadingProviderMock = ({ children }) => (
  <LoadingProvider
    value={{
      loading: false,
      setLoading: jest.fn()
    }}
  >
    {children}
  </LoadingProvider>
);

export { LoadingProviderMock };
