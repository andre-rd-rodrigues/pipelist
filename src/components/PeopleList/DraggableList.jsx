import React from "react";
import ListRow from "components/PeopleList/ListRow";
import { Droppable, DragDropContext, Draggable } from "react-beautiful-dnd";

const DraggableList = ({ list, setList, onPersonSelect }) => {
  const handleOnDragEnd = (result) => {
    if (!result.destination) return;

    const newOrderedList = Array.from(list);
    const [reorderedItem] = newOrderedList.splice(result.source.index, 1);
    newOrderedList.splice(result.destination.index, 0, reorderedItem);

    setList(newOrderedList);
  };

  return (
    <DragDropContext onDragEnd={handleOnDragEnd}>
      <Droppable droppableId="list">
        {(provided) => (
          <div {...provided.droppableProps} ref={provided.innerRef}>
            {list?.map((person, index) => (
              <Draggable
                key={person.id}
                draggableId={person?.id.toString()}
                index={index}
              >
                {(provided) => (
                  <ListRow
                    innerRef={provided.innerRef}
                    provided={provided}
                    onPersonSelect={onPersonSelect}
                    person={person}
                  />
                )}
              </Draggable>
            ))}
            {provided.placeholder}
          </div>
        )}
      </Droppable>
    </DragDropContext>
  );
};

export default DraggableList;
