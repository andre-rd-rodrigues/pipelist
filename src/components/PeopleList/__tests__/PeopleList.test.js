import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import { LoadingProviderMock } from "mocks/LoadingProviderMock";
import reactTestRenderer from "react-test-renderer";
import PeopleList from "../PeopleList";

jest.mock("react-beautiful-dnd", () => ({
  Droppable: ({ children }) =>
    children(
      {
        draggableProps: {
          style: {}
        },
        innerRef: jest.fn()
      },
      {}
    ),
  Draggable: ({ children }) =>
    children(
      {
        draggableProps: {
          style: {}
        },
        innerRef: jest.fn()
      },
      {}
    ),
  DragDropContext: ({ children }) => children
}));

describe("People list", () => {
  beforeAll(() => {
    console.error = jest.fn();
  });

  it("should render correctly", () => {
    const wrapper = reactTestRenderer.create(
      <LoadingProviderMock>
        <PeopleList />
      </LoadingProviderMock>
    );

    expect(wrapper).toMatchSnapshot();
  });
  it("should open PersonModal when person clicked", async () => {
    render(
      <LoadingProviderMock>
        <PeopleList />
      </LoadingProviderMock>
    );

    const person = await screen.findByText(/Mila Benson/i);

    fireEvent.click(person);

    expect(screen.getByText(/Person Information/i)).toBeInTheDocument();
  });
  it("should open AppPersonModal when 'Add Person' button is clicked", () => {
    render(
      <LoadingProviderMock>
        <PeopleList />
      </LoadingProviderMock>
    );

    const addPersonButton = screen.getByTestId(/add_person_button/i);

    fireEvent.click(addPersonButton);

    expect(screen.getByText(/First Name*/i)).toBeInTheDocument();
  });
});
