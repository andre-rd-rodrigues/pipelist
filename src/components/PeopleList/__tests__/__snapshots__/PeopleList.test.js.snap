// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`People list should render correctly 1`] = `
<div
  className="list"
>
  <h1>
    People's
     List
  </h1>
  <hr />
  <div
    id="list_section"
  >
    <div
      className="list_add_person"
    >
      <button
        className="appButton"
        color="green"
        data-testid="add_person_button"
        label="Add Person"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": undefined,
            "border": false,
            "color": "white",
          }
        }
      >
        Add Person
      </button>
    </div>
    <div />
  </div>
</div>
`;
